packages/services/src/kernel/default.ts

```diff
@@ -1256,13 +1258,51 @@ export class KernelConnection implements Kernel.IKernelConnection {
 
     if (this.status !== 'dead') {
       if (connectionStatus === 'connected') {
+        let restarting = false;
+        if (this._kernelSession === RESTARTING_KERNEL_SESSION) {
+          // Reset kernelSession on new connections,
+          // so we can start sending messages.
+          // Otherwise, requests are queued forever until we get a ws message
+          // that we did not ask for.
+          this._kernelSession = '';
```

Comment:
I wouldn't need this on/off toggle if we had `requestKernelInfo(skipTheQueue)` or some such, but it was more layers of APIs than I was comfortable modifying.

Reply:
Agreed that this is a less elegant but also less risky change. Another thing we could do is always let requestKernelInfo requests through if the websocket is connected, i.e., never queue them in sendMessage based on the restarting parameter.